### PR TITLE
AA-404 Adding Stakemanager, unskipping staked validation tests

### PR DESCRIPTION
**Pull request by shahafn**

### CODE REVIEW

File: tests/rip7560/test_validation_rules.py [hunk 1]
```diff
@@ -98,7 +111,7 @@ def entity_to_contract_name(entity):
     return None
 
 
-def get_build_func(entity, rule):
+def get_build_func(entity, rule, assert_func, stake_manager):
```
Comment (drortirosh) [inline tests/rip7560/test_validation_rules.py:114]: it was ugly, and it only gets worse
there is now a lot of code in creating the tests. it was supposed to be parameterised table with no logic

File: tests/rip7560/test_validation_rules.py [hunk 2]
```diff
@@ -110,22 +123,16 @@ def get_build_func(entity, rule):
         # Not implemented yet
         pass
     if rule.find("init_code") > 0:
-        build_func = with_initcode(build_func)
+        # STO-022 factory must be staked
+        if entity == SENDER and assert_func is assert_ok:
```
Comment (drortirosh) [inline tests/rip7560/test_validation_rules.py:127]: I can't say I understand this logic.
I don't see a code where you use unstaked factory - whcih means there are test scenarios you can't test

File: tests/rip7560/conftest.py [hunk 3]
```diff
@@ -72,17 +72,21 @@ def tx_7560(wallet_contract):
 
 @pytest.fixture(scope="session")
 def nonce_manager(w3):
+    artifact_path = "/../../@rip7560/artifacts/contracts/predeploys/NonceManager.sol/NonceManager.json"
+    return contract_from_artifact(w3, artifact_path, CommandLineArgs.nonce_manager)
+
+
+@pytest.fixture(scope="session")
+def stake_manager(w3):
+    artifact_path = "/../../@rip7560/artifacts/contracts/predeploys/StakeManager.sol/StakeManager.json"
```
Comment (drortirosh) [inline tests/rip7560/conftest.py:81]: for tests we usually compile contracts from sol. why here depend on precompiled json ? it requires an ugly "prepare" step

File: pyproject.toml [hunk 4]
```diff
@@ -31,7 +31,7 @@ build-backend = "pdm.pep517.api"
...
 update-deps-remote = {shell = "git submodule update --init --recursive --remote && cd @account-abstraction && yarn && yarn compile &&  cd ../spec && yarn && yarn build && cd ../@rip7560 && yarn && yarn compile-hardhat"}
 test = "pytest --tb=short -rA -W ignore::DeprecationWarning --url http://localhost:3000/rpc --entry-point 0x0000000071727De22E5E9d8BAf0edAc6f37da032  --ethereum-node http://127.0.0.1:8545/ tests/single"
-test-rip7560 = "pytest --tb=short -rA -W ignore::DeprecationWarning --url http://localhost:3000/rpc --entry-point 0x0000000071727De22E5E9d8BAf0edAc6f37da032 --nonce-manager 0x63f63e798f5F6A934Acf0a3FD1C01f3Fac851fF0  --ethereum-node http://127.0.0.1:8545/ tests/rip7560"
+test-rip7560 = "pytest --tb=short -rA -W ignore::DeprecationWarning --url http://localhost:3000/rpc --entry-point 0x0000000071727De22E5E9d8BAf0edAc6f37da032 --nonce-manager 0x63f63e798f5F6A934Acf0a3FD1C01f3Fac851fF0  --stake-manager 0x5271A07b4516A6408E27395941b3D8FC04d96353 --ethereum-node http://127.0.0.1:8545/ tests/rip7560"
```
Comment (drortirosh) [inline pyproject.toml:34]: either entrypoint, or stakemanager/nonceManager. not all 3.

File: tests/rip7560/test_validation_rules.py (hunk 2 above)
Comment (shahafn) [reply tests/rip7560/test_validation_rules.py:127]: Almost all tests use unstaked factory

File: tests/rip7560/conftest.py (hunk 3 above)
Comment (shahafn) [reply tests/rip7560/conftest.py:81]: Because these are predeployed contracts...

File: pyproject.toml (hunk 4 above)
Comment (shahafn) [reply pyproject.toml]: Removed

File: tests/rip7560/test_validation_rules.py (hunk 2 above)
Comment (drortirosh) [reply tests/rip7560/test_validation_rules.py:127]: what I mean: I look at the list of test cases which describe the preconditions for each test.
from this code I understand that sometimes the factory is staked, and sometime it doesn't (that is, when the entity-under-test is not the factory itself)
but from a test case itself, I can't tell the "preconditions"

**Review by forshtat — APPROVED** [review]

**Pull request merged**